walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,87 @@
 
 package com.google.sps;
 
+import java.sql.Time;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
+  private static final Collection<Event> NO_EVENTS = Collections.emptySet();
+  private static final Collection<String> NO_ATTENDEES = Collections.emptySet();
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> ranges = new ArrayList<>();
+    if((events == NO_EVENTS&& request.getDuration()<=TimeRange.WHOLE_DAY.duration()) 
+      || request.getAttendees() == NO_ATTENDEES){
+      ranges.add(TimeRange.WHOLE_DAY);
+      return ranges;
+    }else if(request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+      return ranges;
+    }
+    Collection<String> attendees = request.getAttendees();
+    for(Event event: events){
+      Collection<String> eAttendees = event.getAttendees();
+      for(String attendee: attendees){
+        if(eAttendees.contains(attendee)||eAttendees.isEmpty()){
+          ranges.add(event.getWhen());
```

Comment:
I don't understand what is happening here. If eAttendees.contains(attendee), then we have a conflict and cannot schedule the meeting at this time.

ranges is supposed to contain the set of times the work (at least as used above), but you are adding times that don't work? Looking at your code below, it seems you are using it a bit differently from here on. I think your code would be much easier to follow ranges had a single interpretation. If you need to make a list of time ranges that don't work, create a new local variable.